sdk/core/Azure.Core/src/NullableResponseOfT.cs

```diff
@@ -23,7 +23,7 @@ public abstract class NullableResponse<T>
         /// <summary>
         /// Gets the value returned by the service. Accessing this property will throw if <see cref="HasValue"/> is false.
```

Comment:
Do we need to update the docs to reflect the new behavior?

Reply:
I believe this is still correct. Ex:
https://github.com/Azure/azure-sdk-for-net/blob/0c691e2ee6e0051047f919d23303255a828616b9/sdk/core/Azure.Core/src/Shared/NoValueResponseOfT.cs#L22-L28

Reply:
Hm, perhaps I'm confused.  I thought in some cases accessing the value would just return a `null` value?  Is that not correct?

Reply:
It's up to the implementor of the type - but the doc comment sounds correct to me regardless of this. 

i.e. an implementor could choose to return `true` for `HasValue` and `null` for `Value`.  Perhaps I am misunderstanding what change you are proposing to the comment though.

Reply:
> For example, the customer issue is speaking to the second item - HasValue on `Reponse<T>` is true, but Value is null. Once we make HasValue false when Value is null, it won't throw if you access Value when HasValue is false, so the documentation becomes incorrect.

If we are referring to the few instances of incorrect usage of this type, then yes. I think it remains correct for the vast majory of cases.

> I'm also interested in the case where we would return true for HasValue and null for Value. Is this a real scenario? When would we do this? 

There should be no reason for this to occur, but technically possible.

If it's valid for `NullableResponse<T>`, is it also valid for `Reponse<T>`? This seems problematic for me to change the semantics in an inherited type, which is where I wonder if it would be better just to [EBN] HasValue on `Response<T>` to avoid further issues in the future. Does it address a customer scenario in `Response<T>`, or was it added just because we wanted `NoValueResponse<T>` to be able to implement `NullableResponse<T>` ?

I think to `[EBN]` would not address the problem at all since the implied value would still always be true.

Reply:
> I think to [EBN] would not address the problem at all since the implied value would still always be true.

I had been thinking we would still change the implementation of `HasValue` in `Response<T>` consistent with what you have in this PR, but remove the other API changes, including the one that would be a breaking change to Tables customers.  

It looks like `HasValue` is already [EBN]'ed on `Response<T>` actually, so we wouldn't need to add that!

So if we went that route, the question would just be whether the change to `NullableResponse<T>` is still needed if `Value` can still be null even without `T?`.